### PR TITLE
Add manifest edit command

### PR DESCRIPTION
**Pull request by listx**
Allows editing manifests programmatically. Addresses #190 

This builds on #208 and #209; any changes to those PRs will necessitate a force-push here. Putting this up for now to encourage discussion.

Meanwhile, I'll add some unit tests.

/cc @justaugustus

### CODE REVIEW

**Comment by listx** [comment]
I suggest checking out the README commit; I did not add in support for the requested `--prod-tag` flag yet; that can come later in a separate PR.

**Comment by listx** [comment]
Rebased to get rid of commits from #209.

**Comment by listx** [comment]
Added some unit tests.

**Comment by justaugustus** [comment]
I should be able to review this next week! :)

**Comment by listx** [comment]
@justaugustus Ping?

**Comment by justaugustus** [comment]
@listx -- Assigning some Release Managers to get you some reviews ahead of mine. Something came up last week that's going to have most of my attention this week. Will review when I can though.

/assign @tpepper @saschagrunert @cpanato @hasheddan

File: cmd/cip-mm/main.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	if err := opt.Validate(); err != nil {
+		return err
+	}
+
+	err := reg.GrowManifest(ctx, opt)
+	if err != nil {
+		return err
+	}
+
+	return nil
```
Comment (saschagrunert) [inline cmd/cip-mm/main.go:95]: I suggest in wrapping the errors for better tracing-messages. If we do not want to do something like this, then we could return the result directly here:
```suggestion
	return reg.GrowManifest(ctx, opt)
```

**Comment by listx** [comment]
Applied @saschagrunert's suggestion. Rebased because one of the commits for linting is noise now (was applied in #215).

**Comment by listx** [comment]
@justaugustus Friendly ping? :)

**Comment by justaugustus** [comment]
Thanks Linus (and sorry for the wait)!
/lgtm
/approve
/hold cancel

**Pull request merged**